src/ray/common/client_connection.cc

```diff
@@ -47,7 +47,13 @@ Status ConnectSocketRetry(local_stream_socket &socket, const std::string &endpoi
     if (!ec) {
       break;
     }
-    if (num_attempts > 0) {
+    if (0 < num_attempts && num_attempts < num_retries / 2) {
```

Comment:
nit: could you keep it as `num_attempts > 0`.
Thanks for addressing this :)

Reply:
At risk of starting a flame war, this is soo much closer to `0 < num_attempts < num_retries / 2`. Nevertheless, I've changed it to just never error log :p